File: src/amtsub/sources/rapiddns.py
import requests
from bs4 import BeautifulSoup

class RapidDNS:
	async def search(self, domain, timeout):
		try:
			url = 'https://rapiddns.io/subdomain/' + domain

			response = requests.get(url, timeout = timeout)

			body = response.text

			soup = BeautifulSoup(body, 'html.parser')

			tds = soup.find_all('td')

			subdomains = []

			for td in tds:
				subdomain = td.text

				subdomains.append(subdomain)

			anchors = soup.find_all('a')

			for anchor in anchors:
				href = anchor.href

				if href:
					subdomain = href.split('/')[2].split('#')[0]

					subdomains.append(subdomain)

			return subdomains
		except:
			pass

	def get_name(self):
		return 'RapidDNS'

File: src/amt/sub/sources/rapiddns/rapiddns.go
package rapiddns

import (
	"io"
	"time"
	"net/http"
)

import (
	"amt/utils/strutils"
)

type RapidDNS struct {}

func (self RapidDNS) Search(domain string, timeOut time.Duration) ([]string, error) {
	client := http.Client {
		Timeout: timeOut,
	}

	url := "https://rapiddns.io/subdomain/" + domain

	response, err := client.Get(url)

	if err != nil {
		return []string{}, err
	}

	defer response.Body.Close()

	body, err := io.ReadAll(response.Body)

	if err != nil {
		return []string{}, err
	}

	expression := "[0-9a-z-.]+" + domain

	subdomains := strutils.FindAll(string(body), expression)

	return subdomains, nil
}

func (self RapidDNS) GetName() string {
	return "RapidDNS"
}